docs/devs/e2e-tests.md

```diff
@@ -0,0 +1,194 @@
+```
+npx playwright test openmrs-senaite-flows
+```
+
+### Analyzing a sample test case
+
+Again, following our example of the data flows between OpenMRS and SENAITE, let us verify that ordering a lab test for an OpenMRS patient does create the corresponding SENAITE client with an analysis request. Let us look at the sample test case code below:
+
+```javascript
+import { test, expect } from '@playwright/test';
+import { HomePage } from '../utils/functions/testBase';
+import { patientName } from '../utils/functions/testBase';
+
+let homePage: HomePage;
+
+test.beforeEach(async ({ page }) => {
+  homePage = new HomePage(page);
+  await homePage.initiateLogin();
+  await expect(page).toHaveURL(/.*home/);
+  await homePage.createPatient();
+  await homePage.startPatientVisit();
+});
+
+test('Ordering a lab test for an OpenMRS patient creates the corresponding SENAITE client with an analysis request.', async ({ page }) => {
+  
+  // setup
+  homePage = new HomePage(page);
+  await homePage.goToLabOrderForm();
+
+  // replay
+  await page.click('button:has-text("Add")');
+  await page.selectOption('#tab select', '857AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+  await homePage.saveLabOrder();
+
+  // verify
+  await homePage.goToSENAITE();
+  await expect(page).toHaveURL(/.*senaite/);
+  await homePage.searchClientInSENAITE();
+  const clientName = `${patientName.firstName} ${patientName.givenName}`;
+  const client = await page.$('table tbody tr:nth-child(1) td.contentcell.title div span a:has-text("' + clientName + '")');
+  await expect(client).toBeVisible();
+});
+
+test.afterEach(async ({ page }) => {
+  homePage = new HomePage(page);
+  await homePage.deletePatient();
+  await page.close();
+});
+```
+
+We observe that the test structure is broken down between a **setup**, the actual **test case** and a **cleanup**:
+
+**Test Setup**: Before the actual test, we perform some preliminary actions: logging into Ozone (with SSO), creating a new patient, and starting a visit for the newly created patient.
+
+**Test Case**: The core of each test case follows the _Given-When-Then_ format (commented out here as Setup-Replay-Verification). We highly recommend this structured approach as it clearly delineates the setup of the test environment ("Setup"), the end-user actions performed ("Replay"), and the assertion of outcomes ("Verification"). In our example:
+
+- **Setup**: Navigating to the lab order form, adding a new lab order, and saving it.
+- **Replay**: Navigating to the SENAITE HIS component and searching for the client by name.
+- **Verification**: Verify that the client's name is visible in the clients list.
+
+**Cleanup**: After the test, delete the patient created during the test run and close the browser page.
+
+We Utilize page objects to encapsulate page-specific logic and interactions for better maintainability.
```

Comment:
We need to expand a bit on this to explain better, unless this is just standard Playwright stuff?

Reply:
I think we need to omit this part and concentrate the developer's attention on explanation of the test case structure. The explanation provided should be enough for the developer to write new test cases.